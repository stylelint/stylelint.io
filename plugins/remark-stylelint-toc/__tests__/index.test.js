import remark from "remark";
import stylelintToc from "../";

const content = `bbb

## Options

### \`"always"\`

ccc

### \`"never"\`

ddd

## Optional secondary options

### \`except: ["after-comment", "after-declaration", "first-nested"]\`

#### \`"after-comment"\`

eee

#### \`"after-declaration"\`

fff`;

const contentShort = `bbb

## Options

### \`"always"\`

ccc
`;

test(`don't generate TOC`, () => {
  expect(run(content)).toMatchSnapshot();
});

test(`generate on a rule page before Options`, () => {
  expect(run(content, "RulePage")).toMatchSnapshot();
});

test(`generate on a rule page if comment exist`, () => {
  expect(
    run(`# aaa\n\n<!-- TOC -->\n\n## ab\n\n${content}`, "RulePage")
  ).toMatchSnapshot();
});

test(`generate on any page if comment exist`, () => {
  expect(run(`# aaa\n\n<!-- TOC -->\n\n## ab\n\n${content}`)).toMatchSnapshot();
});

test(`don't generate if TOC is only one level deep in a rule page`, () => {
  expect(run(contentShort, "RulePage")).toMatchSnapshot();
});

test(`generate on a rule page if Options has more than one section`, () => {
  expect(
    run(`${contentShort}\n### \`"never"\`\n\nddd`, "RulePage")
  ).toMatchSnapshot();
});

test(`generate if TOC comment before h2 heading`, () => {
  expect(
    run(`
bbb

<!-- TOC -->

## ccc

ddd
    `)
  ).toMatchSnapshot();
});

test(`generate if TOC comment before h3 heading`, () => {
  expect(
    run(`
bbb

## ccc

<!-- TOC -->

### ddd

eee
    `)
  ).toMatchSnapshot();
});

function run(content, layout) {
  return remark()
    .use(stylelintToc, { layout })
    .process(content).contents;
}
