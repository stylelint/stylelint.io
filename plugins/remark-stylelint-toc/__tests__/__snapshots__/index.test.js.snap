// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`don't generate TOC 1`] = `
"bbb

## Options

### \`\\"always\\"\`

ccc

### \`\\"never\\"\`

ddd

## Optional secondary options

### \`except: [\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"]\`

#### \`\\"after-comment\\"\`

eee

#### \`\\"after-declaration\\"\`

fff
"
`;

exports[`don't generate if TOC is only one level deep in a rule page 1`] = `
"bbb

## Options

### \`\\"always\\"\`

ccc
"
`;

exports[`generate if TOC comment before h2 heading 1`] = `
"bbb

-   [ccc](#ccc)

## ccc

ddd
    
"
`;

exports[`generate if TOC comment before h3 heading 1`] = `
"bbb

## ccc

-   [ddd](#ddd)

### ddd

eee
    
"
`;

exports[`generate on a rule page before Options 1`] = `
"bbb

-   [Options](#options)

    -   [\\"always\\"](#always)
    -   [\\"never\\"](#never)

-   [Optional secondary options](#optional-secondary-options)

    -   [except: \\\\[\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"\\\\]](#except-after-comment-after-declaration-first-nested)

        -   [\\"after-comment\\"](#after-comment)
        -   [\\"after-declaration\\"](#after-declaration)

## Options

### \`\\"always\\"\`

ccc

### \`\\"never\\"\`

ddd

## Optional secondary options

### \`except: [\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"]\`

#### \`\\"after-comment\\"\`

eee

#### \`\\"after-declaration\\"\`

fff
"
`;

exports[`generate on a rule page if Options has more than one section 1`] = `
"bbb

-   [Options](#options)

    -   [\\"always\\"](#always)
    -   [\\"never\\"](#never)

## Options

### \`\\"always\\"\`

ccc

### \`\\"never\\"\`

ddd
"
`;

exports[`generate on a rule page if comment exist 1`] = `
"# aaa

-   [ab](#ab)

-   [Options](#options)

    -   [\\"always\\"](#always)
    -   [\\"never\\"](#never)

-   [Optional secondary options](#optional-secondary-options)

    -   [except: \\\\[\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"\\\\]](#except-after-comment-after-declaration-first-nested)

        -   [\\"after-comment\\"](#after-comment)
        -   [\\"after-declaration\\"](#after-declaration)

## ab

bbb

## Options

### \`\\"always\\"\`

ccc

### \`\\"never\\"\`

ddd

## Optional secondary options

### \`except: [\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"]\`

#### \`\\"after-comment\\"\`

eee

#### \`\\"after-declaration\\"\`

fff
"
`;

exports[`generate on any page if comment exist 1`] = `
"# aaa

-   [ab](#ab)

-   [Options](#options)

    -   [\\"always\\"](#always)
    -   [\\"never\\"](#never)

-   [Optional secondary options](#optional-secondary-options)

    -   [except: \\\\[\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"\\\\]](#except-after-comment-after-declaration-first-nested)

        -   [\\"after-comment\\"](#after-comment)
        -   [\\"after-declaration\\"](#after-declaration)

## ab

bbb

## Options

### \`\\"always\\"\`

ccc

### \`\\"never\\"\`

ddd

## Optional secondary options

### \`except: [\\"after-comment\\", \\"after-declaration\\", \\"first-nested\\"]\`

#### \`\\"after-comment\\"\`

eee

#### \`\\"after-declaration\\"\`

fff
"
`;
